server/src/main/java/org/apache/iotdb/db/mpp/plan/planner/LogicalPlanner.java

```diff
@@ -423,6 +424,15 @@ public PlanNode visitCountLevelTimeSeries(
           .getRoot();
     }
 
+    @Override
+    public PlanNode visitCountNodes(CountNodesStatement countStatement, MPPQueryContext context) {
+      LogicalPlanBuilder planBuilder = new LogicalPlanBuilder(context);
+      return planBuilder
+          .planChildPathsSchemaSource(countStatement.getPartialPath(), countStatement.getLevel())
+          .planCountMerge()
+          .getRoot();
```

Comment:
How about both sg above and sg below need to count together? like there has sg: root.ln, root.abc.a1.a2.a3 and ts:root.ln.a1.a2.a3. We want count nodes root.** level=3? I think planNodeManagementMemoryMerge() should be used.

Reply:
From my perspective, there is not allowed to have both root.ln and root.ln.a1.a2.a3?

Reply:
Besides, I use get child path in count node, so if there is possible having same path?